libs/core/langchain_core/_api/beta_decorator.py

```diff
@@ -270,7 +270,7 @@ def warn_beta(
             message += f" {addendum}"
 
     warning = LangChainBetaWarning(message)
-    warnings.warn(warning, category=LangChainBetaWarning, stacklevel=2)
```

Comment:
whats this do?

Reply:
It gives better information about which user-side call triggered the warning.
With stacklevel = 2
```
tests/unit_tests/language_models/chat_models/test_rate_limiting.py::test_rate_limit_invoke
  /Users/christophebornet/workspace/langchain/libs/core/langchain_core/_api/beta_decorator.py:87: LangChainBetaWarning: Introduced in 0.2.24. API subject to change.
    warn_beta(
```

With stacklevel = 4
```
tests/unit_tests/language_models/chat_models/test_rate_limiting.py::test_rate_limit_invoke
  /Users/christophebornet/workspace/langchain/libs/core/tests/unit_tests/language_models/chat_models/test_rate_limiting.py:13: LangChainBetaWarning: Introduced in 0.2.24. API subject to change.
    rate_limiter=InMemoryRateLimiter(
```
I believe warn_beta is always called through `@beta` so it makes more sense to warn on the caller of the `@beta` annotated class.
I tried to adjust all stacklevels with this in mind.

Reply:
ooh thats great! yea the beta warnings aren't great atm